python/helpers/jetmetCorrector.py

```diff
@@ -142,21 +234,47 @@ def correctJetAndMET(self, jets, met=None, rho=None, genjets=[], isMC=True, runN
-                    jetReCalibrator = self.jetReCalibratorDATA_BCD
-                elif runNumber >= 276831 and runNumber <= 278801:
-                    jetReCalibrator = self.jetReCalibratorDATA_EF
-                elif runNumber >= 278802 and runNumber <= 280385:
-                    jetReCalibrator = self.jetReCalibratorDATA_G
-                elif runNumber >= 280919 and runNumber <= 284044:
-                    jetReCalibrator = self.jetReCalibratorDATA_H
-                else:
-                    raise RuntimeError("Run %d out of range" % runNumber)
+		if(self.Year==2016):
+			if runNumber >= 272007 and runNumber <= 276811:
+			    jetReCalibrator = self.jetReCalibratorDATA_BCD
+			elif runNumber >= 276831 and runNumber <= 278801:
+			    jetReCalibrator = self.jetReCalibratorDATA_EF
+			elif runNumber >= 278802 and runNumber <= 280385:
+			    jetReCalibrator = self.jetReCalibratorDATA_G
+			elif runNumber >= 280919 and runNumber <= 284044:
+			    jetReCalibrator = self.jetReCalibratorDATA_H
+			else:
+			    raise RuntimeError("Run %d out of range" % runNumber)
+
+                elif(self.Year==2017): #Run numbers for each era taken from this presentation (slide 21): https://indico.cern.ch/event/775563/contributions/3224847/attachments/1759380/2853926/20181126_JERC_Status_2016_2017.pdf
+			if runNumber >= 297046 and runNumber <= 299329:
+			    jetReCalibrator = self.jetReCalibratorDATA_B
+			elif runNumber >= 299368 and runNumber <= 302029:
+                            jetReCalibrator = self.jetReCalibratorDATA_C
+                        elif runNumber >= 302030 and runNumber <= 304797:
+                            jetReCalibrator = self.jetReCalibratorDATA_DE
+                        elif runNumber >= 305040 and runNumber <= 306462:
+                            jetReCalibrator = self.jetReCalibratorDATA_F
+                        else:
+                            raise RuntimeError("Run %d out of range" % runNumber)
+
+                elif(self.Year==2018): #Run numbers for each era taken from the twiki: https://twiki.cern.ch/twiki/bin/view/CMS/PdmV2018Analysis
+                        if runNumber >= 315252 and runNumber <= 316995:
+                            jetReCalibrator = self.jetReCalibratorDATA_A
+                        elif runNumber >= 317080 and runNumber <= 319310:
+                            jetReCalibrator = self.jetReCalibratorDATA_B
+                        elif runNumber >= 319337 and runNumber <= 320065:
+                            jetReCalibrator = self.jetReCalibratorDATA_C
+                        elif runNumber >= 320673 and runNumber <= 325175:
+                            jetReCalibrator = self.jetReCalibratorDATA_D
+                        else:
+                            raise RuntimeError("Run %d out of range" % runNumber)
+
             for j in jets:
                 newpt = jetReCalibrator.correct(j, rho)
                 j.mass = newpt / j.pt * j.mass
                 j.pt = newpt
-
```

Comment:
Just to confirm: Does this still work with the **latest** official NanoAOD-tools? I remember the JME correction part has been changed quite a bit in the past few months.

Reply:
@pkontaxa You can check this by setting up a new area, pull the latest NanoAOD-tools and your branch (including these changes in this PR), and test if it runs.